cli/goals_principles.md

```diff
@@ -0,0 +1,74 @@
+# Abstract
+
+This document will outline the goals of creating a mobile CLI for K8s / OpenShift. It will also outline the technical considerations and design principles that should be used to guide the development of not only the mobile CLI, but any other mobile service CLI.
+
+
+
+# Goals
+
+The key goal of the mobile cli, is to provide a simple and intuitive interface to enable and assist developers to adopt and use Open Source mobile services on top of Kubernetes and OpenShift in order to solve their mobile project requirements.
+
+
+# General design principles and UX considerations
+
+## Output
+
+As this is a CLI, it is important that it remains a programable grep and pipe friendly interface to allow scripting and execution from external programs. This generally means it should write its output to stdOut. Using a format like JSON allows it to be piped to tools such as [jq](https://stedolan.github.io/jq/) for further use.
+
+However it should default to presenting its output in a human readable format such as a table structure:
+```
++-----------------------+-----------------------------+
+|         NAME          |             ID              |
++-----------------------+-----------------------------+
+| dh-fh-sync-server-apb | dh-fh-sync-server-apb-59dkt |
++-----------------------+-----------------------------+
+```
+
+## Arguments and Flags
+	
+- An argument represents a required parameter needed in order to complete the requested action. 
```

Comment:
Going by the example, is it worth mentioning that arguments have a required order too?

Reply:
No I think that is covered under the user instruction section